src/amo/components/CollectionAddAddon/index.js

```diff
@@ -82,10 +109,17 @@ export class CollectionAddAddonBase extends React.Component<InternalProps> {
   }
 
   componentWillUnmount() {
+    this.resetMessageStatus();
+  }
+
+  resetMessageStatus = () => {
```

Comment:
`resetMessages`, maybe? I suggest that because clearing the fader timeout doesn't really have to do with status.